### DISCUSSION TITLE
Mahsa Amini Heat Map | نقشه حرارتی شهرهای به پاخاسته

### DISCUSSION THREAD

**Discussion by amastaneh** [WeAreMahsaAmini/Core#96] [Ideas]
I came up with this idea last week and started to develop it.

## Wishes
- We can extract the names of cities from Twitter messages tagged with Mehsa Amini's hashtags and map them on the Heat Map to determine which cities are influenced.

## Appritciations
- Twitter granted me access to two million free tweets after I requested more access for this purpose.

## Blocker
- Is it possible that this idea will work against itself and be used by the dictator? I really don't know!
- I tried to design the user interface based on the COVID-19 Heat Map, but I am not sure how useful it is.

![image](https://user-images.githubusercontent.com/6195199/192693285-403bf8c7-76d6-4630-b37a-360adf0dc491.png)

## Simillar
- [by JeffShomali](https://github.com/JeffShomali) / [discussions/37](https://github.com/WeAreMahsaAmini/Core/discussions/37#discussion-4426192) 
- [by united4iran](https://iranrally.ushahidi.io/views/map)

**Comment by amastaneh** [comment]
Twitter approved access to 2 million tweets for free.

<img src="https://user-images.githubusercontent.com/6195199/192831581-1c54a5b0-e1bf-46bd-97cd-415f17ab4c22.png" width="600" />
<img src="https://user-images.githubusercontent.com/6195199/192831771-9d269c19-fad7-42ca-924b-4bbeef21a4fb.png" width="600" />

Would you help with blockers?

**Reply by Pouria-007** [reply]
Great news!

**Comment by Pouria-007** [comment]
This is a brilliant idea! I will support it.

**Comment by theStratus** [comment]
Nice idea. It is also implemented here. 
https://github.com/WeAreMahsaAmini/Core/discussions/37#discussioncomment-3752142
you may contribute for integration. Anyway, if it is real time, then it may act against people. However, it may be a list of previous protest that can be used for further planning of protests.

**Reply by amastaneh** [reply]
Honestly, as I mentioned in Blocker, this idea can work against itself. \
This statistic (live or previous) can be used by Murderer forces to manage their forces.

**Reply by theStratus** [reply]
I think if it is not live, it may be safe.

**Reply by amastaneh** [reply]
Is it possible for the murderous forces to use previous information to organize future suppressions?\
That is my only concern.

**Reply by ptkx132** [reply]
maybe we can provide statistics only about murderers. based on observation of people. and also marked the sensitive places ( where they keep arms, garages, suppliers ... )

**Comment by BarayeAzadi** [comment]
What if we change the heat map to show the location of polices rather than the people? This way the chances of helping the police is much less and people know what's going on in the city

**Reply by amastaneh** [reply]
Do you have any idea for sources and how we can rely on?

**Reply by BarayeAzadi** [reply]
considering the filtering and low VPN access, I doubt tweeter going to be useful. we need to provide a reporting system that cannot be tracked and cannot be blocked... we can reuse this system for many other purposes. It might be too ambitious but that's the way to do it.

**Reply by MarjanDavoodi** [reply]
I like this idea! Can we start by simply mapping police stations and Sepah bases? It is far from being useful but we can gradually add more data, from social media reports, or a more sophisticated reporting system like you suggested.

**Reply by BarayeAzadi** [reply]
Agreed. @amastaneh can we start it as above?

**Comment by MarjanDavoodi** [comment]
We have evidence that Twitter data has been used for protest policing in the past (e.g., https://theintercept.com/2020/07/09/twitter-dataminr-police-spy-surveillance-black-lives-matter-protests/). However, there're still ways to protect people AND benefit from collecting this data; for instance, by releasing the aggregate data first (e.g., city level) and releasing neighborhood location later on, if it's available...

**Reply by amastaneh** [reply]
Is it possible for the murderous forces to use previous information to organize future suppressions?
That is my only concern.

**Reply by MarjanDavoodi** [reply]
This is a real concern that can be discussed for hours. I am somewhat familiar with the literature that addresses cases in which data was used for protest prediction and policing in the US context. However, these findings are generally inconclusive, considering that we don’t really know if governments actually invest in such sophisticated surveillance tools. 
In the end, I would say protest event data collection is a very common practice, both for good and evil. My main collaborator who has studied protest policing for decades has collected some major protest datasets as well! We can use good examples (such as the Crowd Counting Consortium dataset) to make sure we are on the right track. 

I am contacting someone who has extensively worked on the intersection of NLP, protest event data collection, and algorithmic fairness (plus other related ethical concerns) at Google. I try to get more input as we move forward and need to make more technical decisions, with this concern in mind.

**Reply by MarjanDavoodi** [reply]
I just had a meeting with the person I mentioned! She is trying to put together a paper outlining the actions needed to take to minimize the likelihood of protest policing and predictive policing. At this point, it seems like protest event data collectors haven't cared enough to offer us some solutions to prevent this issue! 

Alex's concern was less about location than people's names or organizations being somehow identified, considering that it is obvious that people gather in some of the significant locations of protest such as Enghelab Square in Tehran. However, I would say since this wave of protests is more diffused in neighborhoods, targeting based on the mere location variable is still a valid concern. So, the city versus neighborhood-level data (in case we get sophisticated enough to extract it) is a major decision. 

In the end, this might come to us to clarify what we try to achieve with this data. Are we trying to help people find accurate enough protest spots to join? That would likely invite some brutal forces that we don't want to be there! But if we decide to not even release the live data, we need to be clear about our goals and make these decisions accordingly.

**Comment by MarjanDavoodi** [comment]
Take a look at this paper if you get a chance: https://journals.sagepub.com/doi/full/10.1177/0081175019860244

Here, they use a Chinese microblogging platform to extract protest event data. This is certainly doable but it requires external validation to produce robust data. It is challenging in the case of Iran because there are not much data available on it and it requires being creative to find external metrics for validation.

**Comment by matinirani** [comment]
I like the idea. Any update? any help needed???

**Reply by MarjanDavoodi** [reply]
I have my Comps exams in 9 days:( Someone else should start planning, coding, etc. and I can jump in ASAP! Thanks:)

**Reply by amastaneh** [reply]
به این نتیجه رسیدم این ایده می‌تونه برای توزیع نیروهای سرکوب مفید واقع بشه و برای همین متوقف شد.
